src/lib.rs

```diff
@@ -644,6 +645,40 @@ where
     }
 }
 
+/// A panic payload indicating that a salsa revision was canceled.
+#[derive(Debug)]
+pub struct Canceled {
+    _private: (),
+}
```

Comment:
```suggestion
#[derive(Debug)]
#[non_exhaustive]
pub struct Canceled;
```

I think we can do this now

Reply:
Oh that's cool

Reply:
Not sure if it actually works, but seems like it should?